File: 2021/May/Day 20/base7.py
# Source : https://leetcode.com/problems/base-7/submissions/
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if(num==0):
            return "0";
        s= "";

        if(num<0):
            sign ="-"
        else:
            sign = "";
        num = abs(num);

        while(num):
            s+=str(int(num)%7);
            num = int(num/7);
        s+=sign;
        return s[::-1]

File: 2023/January/Day 19/base7.cpp
// Source : https://leetcode.com/problems/base-7/submissions/
class Solution {
public:
    string convertToBase7(int num) {
        string out="";
        int VERBOSE=0;
        if(num==0)return "0";
        int sign=1;
        if(num<0)sign=-1;
        num = abs(num);
        
        while(num)
        {
            out+=to_string(num%7);
            num/=7;
            
        }
        reverse(out.begin(),out.end());
        if(VERBOSE)
        cout<<out;
        if(sign==-1)
        {
            out="-"+out;
        }
        return (out);
    }
};